qiita_db/handlers/analysis.py

```diff
@@ -56,15 +59,34 @@ def get(self, analysis_id):
         -------
         dict
             The contents of the analysis keyed by sample id
+
+        Notes
+        -----
+        This response needed to be broken in chunks because we were hitting
+        the max size of a respose: 2G; based on: https://bit.ly/3CPvyjd
         """
+        chunk_len = 1024 * 1024 * 1  # 1 MiB
+
         with qdb.sql_connection.TRN:
             a = _get_analysis(analysis_id)
             mf_fp = qdb.util.get_filepath_information(
                 a.mapping_file)['fullpath']
-            response = None
             if mf_fp is not None:
                 df = qdb.metadata_template.util.load_template_to_dataframe(
                     mf_fp, index='#SampleID')
-                response = df.to_dict(orient='index')
+                response = dumps(df.to_dict(orient='index'))
 
-        self.write(response)
+                crange = range(chunk_len, len(response)+chunk_len, chunk_len)
+                for i, (win) in enumerate(crange):
+                    chunk = response[i*chunk_len:win]
+                    try:
+                        self.write(chunk)
+                        await self.flush()
+                    except StreamClosedError:
+                        break
+                    finally:
+                        del chunk
+                        # pause the coroutine so other handlers can run
+                        await gen.sleep(0.000000001)  # 1 nanosecond
```

Comment:
wouldn't this be taken care of by the `await` on 84?

Reply:
Nop, in fact, if you check the code in the reference, the magic actually happens here; without this everything goes 🍌  ... at least based on that link.

Reply:
for the large data case, it shouldn't be necessary, but since this code is used on small targets it seems reasonable. it's ashame the `await None` doesn't work. it's possible `yield None` does but it may depend on aspects of decoration that i dont understand